dev/com.ibm.ws.springboot.support_fat/fat/src/com/ibm/ws/springboot/support/fat/TemplateTests.java

```diff
@@ -61,34 +59,44 @@ public static void tearDown() throws Exception {
         machine.setWorkDir(previousWorkDir);
     }
 
-    @Before
-    public void cleanupBeforeRun() throws Exception {
-        // since we are not using the normal LibertyServer class for this server,
-        // we need to make sure to explicitly clean up.  We do this before running
-        // the test in order to preserve the contents on disk.
-        if (LibertyFileManager.libertyFileExists(machine, springBootServerPath)) {
-            LibertyFileManager.deleteLibertyDirectoryAndContents(machine, springBootServerPath);
+    // Test-specific fields
+    private static String[] expectedFeatures;
+    private static String serverName;
+    private static String serverPath;
+
+    @After
+    public void cleanupServer() throws Exception {
+        // Since we are not using the normal LibertyServer class, we need to explicitly
+        // clean up a server. We cleanup before server creation to ensure a server does
+        // not already exist, and then after each test to leave the test environment as
+        // we found it.
+        if (LibertyFileManager.libertyFileExists(machine, serverPath)) {
+            LibertyFileManager.deleteLibertyDirectoryAndContents(machine, serverPath);
         }
     }
 
-    static String[] expectedFeatures;
-
     @Test
     @Mode(FULL)
     public void testCreateServerFromSpringBoot1() throws Exception {
         expectedFeatures = new String[] { "servlet-4.0", "ssl-1.0", "transportSecurity-1.0", "websocket-1.1", "springBoot-1.5" };
+        serverName = "com.ibm.ws.springboot.support.fat.TemplateServer1";
+        serverPath = installPath + "/usr/servers/" + serverName;
 
         createServerFromTemplate("springBoot1", expectedFeatures);
     }
 
     @Test
     public void testCreateServerFromSpringBoot2() throws Exception {
         expectedFeatures = new String[] { "servlet-4.0", "ssl-1.0", "transportSecurity-1.0", "websocket-1.1", "springBoot-2.0" };
+        serverName = "com.ibm.ws.springboot.support.fat.TemplateServer2";
+        serverPath = installPath + "/usr/servers/" + serverName;
 
         createServerFromTemplate("springBoot2", expectedFeatures);
     }
 
     void createServerFromTemplate(String templateName, String[] expectedFeatures) throws Exception {
+        cleanupServer();
```

Comment:
I don't get why you need to do this begore and after for this test sice you're using different servernames. But it seems harmless

Reply:
Sam, before these changes the test left `TemplateServer` in `wlp/usr/servers` after `testCreateServerFromSpringBoot2`.  I thought we should fix that.